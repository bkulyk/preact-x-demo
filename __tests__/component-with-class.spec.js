import React from 'react';
import componentWithClass from '@bitchin/component-with-class';
import { shallow } from './enzyme';
import { it, expect } from './jest';

it('should be a function', () => {
  expect(typeof componentWithClass).toBe('function');
});

it('should create an element with a class name and a type', () => {
  const Comp = componentWithClass('li', 'my-custom-class');
  expect(shallow(<Comp />)).toMatchSnapshot();
});

it('the component should be able set props', () => {
  const Comp = componentWithClass('div', 'my-custom-class');
  expect(shallow(<Comp data-something="someData" className="additional Class" />)).toMatchSnapshot();
});

it('the component should be able set props', () => {
  const Comp = componentWithClass('span', 'my-custom-span');
  expect(shallow(<Comp>child data</Comp>)).toMatchSnapshot();
});
