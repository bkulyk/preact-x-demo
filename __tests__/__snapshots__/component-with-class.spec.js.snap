// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create an element with a class name and a type 1`] = `
<li
  className="my-custom-class"
/>
`;

exports[`the component should be able set props 1`] = `
<div
  className="my-custom-class additional Class"
  data-something="someData"
/>
`;

exports[`the component should have children prop 1`] = `
<span
  className="my-custom-span"
>
  child data
</span>
`;
